skimage/morphology/tests/test_watershed.py

```diff
@@ -489,5 +471,28 @@ def test_incorrect_markers_shape():
         markers = np.ones((5, 7))
         output = watershed(image, markers)
 
+
+def test_incorrect_mask_shape():
+    with pytest.raises(ValueError):
+        image = np.ones((5, 6))
+        mask = np.ones((5, 7))
+        output = watershed(image, markers=4, mask=mask)
+
+
+def test_markers_in_mask():
+    data = blob
+    mask = (data != 255)
+    out = watershed(data, 25, connectivity=2, mask=mask)
+    # There should be no markers where the mask is false
```

Comment:
Why do you call this markers? For me markers and seeds are synonyms, corresponding to marked pixels from which the watershed flooding is done. How about "labeled pixels"? Sorry to be nitpicking about a comment in a test ;-).

Reply:
OK, I think I got it, we're talking about the same thing for markers, you just want to check that no marker was chosen outside the mask, so that all pixels outside the mask in the segmentation are set to zero. OK!